src/PostgREST/App.hs

```diff
@@ -161,438 +113,47 @@ postgrestResponse
   -> Wai.Request
   -> Handler IO Wai.Response
 postgrestResponse conf maybeDbStructure pool time req = do
-  body <- lift $ Wai.strictRequestBody req
-
-  dbStructure <-
-    case maybeDbStructure of
-      Just dbStructure ->
-        return dbStructure
-      Nothing ->
-        throwError Error.ConnectionLostError
-
-  apiRequest@ApiRequest{..} <-
-    liftEither . mapLeft Error.ApiRequestError $
-      ApiRequest.userApiRequest conf dbStructure req body
-
   -- The JWT must be checked before touching the db
-  jwtClaims <- Auth.jwtClaims conf (toS iJWT) time
-
-  let
-    handleReq apiReq =
-      handleRequest $ RequestContext conf dbStructure apiReq
-
-  runDbHandler pool (txMode apiRequest) jwtClaims .
-    Middleware.optionalRollback conf apiRequest $
-      Middleware.runPgLocals conf jwtClaims handleReq apiRequest
+  jwtClaims <- Auth.jwtClaims conf req time
+  body <- lift $ Wai.strictRequestBody req
+  dbStructure <- maybe (throwError Error.ConnectionLostError) pure maybeDbStructure
```

Comment:
Do we need to read the body before getting the dbStructure? I don't think so. Let's swap those 2 lines.

No need to read the body before throwing a `ConnectionLostError`.

Probably makes sense to do:
- `dbStructure`
- `jwtClaims`
- `body`
- `request`

In this order?

Reply:
Yes, that order makes most sense! Fixed.